.github/workflows/openjdk_14_build.yml

```diff
@@ -0,0 +1,140 @@
+name: openjdk_14_build
+on: [push]
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    container: gcr.io/swirlds-registry/cci-openjdk-ghr:14.0.2
+    env:
+      MAVEN_OPTS: -Xmx5200m
+      LC_ALL: C.UTF-8
+      DEBIAN_FRONTEND: noninteractive
+      SLACK_ACCESS_TOKEN: ${{ secrets.SLACK_ACCESS_TOKEN }}
+      SLACK_API_TOKEN: ${{ secrets.SLACK_API_TOKEN }}
+      SONAR_PROJECT_KEY: ${{ secrets.SONAR_PROJECT_KEY }}
+      SONAR_SLACK_CHANNEL: ${{ secrets.SONAR_SLACK_CHANNEL }}
+      SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+    timeout-minutes: 60
+    steps:
+      - run: apt-get -y install software-properties-common
+      - run: add-apt-repository ppa:git-core/ppa
+      - run: apt-get -y install git
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - run: mvn --version
+      - name: Maven Dependency Cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: v3-${{ runner.os }}-mvn-dependencies-${{ hashFiles('pom.xml') }}
+      - name: Compile
+        run: mvn -DskipTests clean compile
+      - name: Unit Test
+        run: mvn -Dsonar.branch.name=${GITHUB_REF_NAME} test verify sonar:sonar
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          files: "*/target/surefire-reports/*.xml"
+      - name: Sonar Check Quality Gate
+        shell: bash
+        id: sonar_gate
+        run: |
+          set -x;
+
+          if [[ "${GITHUB_REF_NAME}" == "main" ]]; then
+            echo "Skipping SonarCloud Quality Gate Enforcement - Branch is Temporarily Excluded (main)........";
+            exit 0;
+          fi;
+
+          REPO_PATH="${PWD}";
+          if [[ ! -d "${REPO_PATH}" ]]; then exit 15; fi;
+
+          SONAR_QG_SCRIPT="${REPO_PATH}/.github/scripts/sonar_check_quality_gate.pl";
+          if [[ ! -f "${SONAR_QG_SCRIPT}" ]]; then exit 38; fi;
+
+          SONAR_REPORT_FILE="${REPO_PATH}/target/sonar/report-task.txt";
+          if [[ ! -f "${SONAR_REPORT_FILE}" ]]; then exit 40; fi;
+
+          export SONAR_TASK_ID="$(cat "${SONAR_REPORT_FILE}" | perl -0777 -e 'while (my $line = <>) { if ($line =~ /ceTaskId=([A-Za-z0-9_-]+)/ig) { print "$1"; } }')";
+          if [[ -z "${SONAR_TASK_ID}" ]]; then exit 41; fi;
+          
+          perl ${SONAR_QG_SCRIPT};
```

Comment:
Why this very complicated perl script instead of the standard sonar maven plugin? The [docs](https://sonarcloud.io/project/configuration?analysisMode=GitHubActions) suggest this when setting up a project. Mirror node workflow example: https://github.com/hashgraph/hedera-mirror-node/blob/f3848377a55d953b7529633585dc2f2cc65c4490/.github/workflows/security.yml#L53

Reply:
Just copy-pasta from circleci config. I’ll see if I can replace.

Reply:
done

Reply:
You can also specify some of the sonar parameters via the [pom.xml](https://github.com/hashgraph/hedera-mirror-node/blob/e10b4579fb25dd1c02a02f64c5b151c07aca38fd/pom.xml#L84)

Reply:
they were already there :-)